### PR TITLE
Add ETL logic for MTB project data outside of the routine QUK17 project

### PR DESCRIPTION
**Pull request by sven1103**

### CODE REVIEW

File: drop-boxes/register-mtb-projects-dropbox/register-mtb-data-dropbox.py [hunk 1]
```diff
@@ -0,0 +1,758 @@
...
+    2.) MTB archive: A ZIP archive containing 6 TSV. These contain information
+        about SNVs, CNVs and metadata. The specification can be found on the mtbparser
+        GitHub repository:
+        https://github.com/qbicsoftware/qbic.mtbparser
+        
+1. Raw data
+The sequencing facilities CeGaT and the human genetics department at UKT only see the QBiC barcode
+of the tumor sample. However, as part of the whole MTB process, during patient registration in openBIS
+an additional sample for PBMC is generated and attached as sample to the patient. It carries
+an unique barcode, that is different from the tumor's one.
```
Comment (wow-such-code) [inline drop-boxes/register-mtb-projects-dropbox/register-mtb-data-dropbox.py]:
```suggestion
a unique barcode, that is different from the tumor's one.
```

File: drop-boxes/register-mtb-projects-dropbox/register-mtb-data-dropbox.py [hunk 2]
```diff
@@ -0,0 +1,758 @@
...
+
+So in this dropbox we query the patient to whome the tumor sample barcode belongs and can access the
```
Comment (wow-such-code) [inline drop-boxes/register-mtb-projects-dropbox/register-mtb-data-dropbox.py]:
```suggestion
So in this dropbox we query the patient to whom the tumor sample barcode belongs and can access the
```

File: drop-boxes/register-mtb-projects-dropbox/register-mtb-data-dropbox.py [hunk 3]
```diff
@@ -0,0 +1,758 @@
...
+
+print(mtbutils.log_stardate("Mtbconverter executable found."))
+
+config = ConfigParser.ConfigParser()
+config.read(PROPERTIES)
+
+api = HttpInvokerUtils.createServiceStub(IApplicationServerApi, config.get('openbis','url') + IApplicationServerApi.SERVICE_URL, 5000)
+ 
+sessionToken = api.login(config.get('openbis','user'), config.get('openbis','password'))
+
```
Comment (wow-such-code) [inline drop-boxes/register-mtb-projects-dropbox/register-mtb-data-dropbox.py]: Why is this needed in the jython script?

File: drop-boxes/register-mtb-projects-dropbox/register-mtb-data-dropbox.py [hunk 4]
```diff
@@ -0,0 +1,758 @@
...
+    space, project = space_and_project(dna_barcode)
+
+    search_service = transaction.getSearchService()
+    sc = SearchCriteria()
+    pc = SearchCriteria()
+    pc.addMatchClause(SearchCriteria.MatchClause.createAttributeMatch(SearchCriteria.MatchClauseAttribute.PROJECT, project));
+    sc.addSubCriteria(SearchSubCriteria.createExperimentCriteria(pc))
+    result = search_service.searchForSamples(sc)
+    print("Found {} samples for project {} in space {}.".format(len(result), project, space))
+    new_rna_sample_barcode = getNextFreeBarcode(project, numberOfBarcodes=len(result))
```
Comment (wow-such-code) [inline drop-boxes/register-mtb-projects-dropbox/register-mtb-data-dropbox.py]: this includes entity barcodes, which will lead to a miscount and future inconsistencies. it is better to check in a while loop if a certain code is already used (by fetching that sample) and incrementing the code as it is done in other ETL scripts.

File: drop-boxes/register-mtb-projects-dropbox/register-mtb-data-dropbox.py [hunk 5]
```diff
@@ -0,0 +1,758 @@
...
+    # Now we have to create a new TEST_SAMPLE with sample type RNA and attach it
+    # to the tumor tissue sample
+    new_rna_sample = transaction.createNewSample("/{space}/{barcode}".format(
+        space=space,
+        barcode=new_rna_sample_barcode), "Q_TEST_SAMPLE")
+    new_rna_sample.setExperiment(transaction.getSearchService().getExperiment(
+        "/{space}/{project}/{project}E{number}".format(
+            space=space,
+            project=project,
+            number=3
```
Comment (wow-such-code) [inline drop-boxes/register-mtb-projects-dropbox/register-mtb-data-dropbox.py]: is this always experiment 3?

File: drop-boxes/register-mtb-projects-dropbox/register-mtb-data-dropbox.py [hunk 6]
```diff
@@ -0,0 +1,758 @@
...
+    tumor_dna_sample_id = qbiccode_f1[0]
+    # Get space and project ids
+    space, project = space_and_project(tumor_dna_sample_id)
+
+    # Create new experiment id
+    experiments = transaction.getSearchService().listExperiments('/{}/{}'.format(space, project))
+    last_exp_id = get_last_exp_id(experiments)
+    new_exp_id = '/{space}/{project}/{project}E{number}'.format(
+        space=space, project=project, number=last_exp_id + COUNTER.newId())
+    new_sample_id = '/{space}/NGS{barcode}-{timestamp}'.format(
```
Comment (wow-such-code) [inline drop-boxes/register-mtb-projects-dropbox/register-mtb-data-dropbox.py]: timestamps in the sample code?

**Review by wow-such-code — CHANGES_REQUESTED** [review]
This is a lot, so I could not have a close look at all of the code, but overall it makes sense.
I think you could and should reuse some code from other ETL scripts, maybe we could even move it to data-model-lib in the process, but that is not critical.

Sample code incrementation should be reworked, I think this will lead to problems otherwise.

File: drop-boxes/register-mtb-projects-dropbox/register-mtb-data-dropbox.py (hunk 4 above)
Comment (sven1103) [reply drop-boxes/register-mtb-projects-dropbox/register-mtb-data-dropbox.py]: I am happy for a url with a cross-reference :)

File: drop-boxes/register-mtb-projects-dropbox/register-mtb-data-dropbox.py (hunk 5 above)
Comment (sven1103) [reply drop-boxes/register-mtb-projects-dropbox/register-mtb-data-dropbox.py]: yup

File: drop-boxes/register-mtb-projects-dropbox/register-mtb-data-dropbox.py (hunk 3 above)
Comment (sven1103) [reply drop-boxes/register-mtb-projects-dropbox/register-mtb-data-dropbox.py]: V3 api usage

File: drop-boxes/register-mtb-projects-dropbox/register-mtb-data-dropbox.py (hunk 6 above)
Comment (sven1103) [reply drop-boxes/register-mtb-projects-dropbox/register-mtb-data-dropbox.py]: my solution at the time to make it unique and allow for multiple runs

**Review by wow-such-code — APPROVED** [review]
Pull the lever Kronk!
